tools/flang1/flang1exe/module.c

```diff
@@ -371,7 +372,11 @@ apply_use_stmts(void)
   save_lineno = gbl.lineno;
 
   if (!gbl.currmod && gbl.internal <= 1) {
-    init_use_tree();
+    for (m_id = FIRST_USER_MODULE; m_id < usedb.avl; m_id++) {
+      if (usedb.base[m_id].scope > sem.scope_level) {
```

Comment:
Nit: Can you add a comment here explaining why removing from the tree is the right action here.